File: src/1832. Check if the Sentence Is Pangram/1832.py
class Solution:
    def checkIfPangram(self, sentence: str) -> bool:
        state = 0
        for c in sentence:
            state |= 1 << (ord(c) - ord('a'))
        return state == (1 << 26) - 1
    def checkIfPangram1(self, sentence: str) -> bool:
        exist = [False] * 26
        for c in sentence:
            exist[ord(c)-ord('a')] = True 
        return all(exist)

if __name__ == "__main__":
    sentence = "thequickbrownfoxjumpsoverthelazydog"
    print(Solution().checkIfPangram(sentence))

File: src/1832. Check if the Sentence Is Pangram/1832.go
func checkIfPangram(sentence string) bool {
	state := 0
	for _, c := range sentence {
		state |= 1 << (c - 'a')
	}
	return state == 1<<26-1
}

func checkIfPangram1(sentence string) bool {
	exist := [26]bool{}
	for _, c := range sentence {
		exist[c-'a'] = true
	}
	for _, b := range exist {
		if !b {
			return false
		}
	}
	return true
}